### PR TITLE
fix(navbar): Improve navbar colors in dark mode

### PR DESCRIPTION
**Pull request by gadenbuie**
This rule removed in this PR only changes the background color of the navbar, which is not enough to fully restyle the navbar in dark mode. It might work in the default cases, but it certainly does not work if users have set `$navbar-dark-bg` or `$navbar-light-bg` to appropriate values, leading to inaccessible values (e.g. white text on a light background).

This PR removes the changing of the navbar background color in dark mode. This is inline with upstream expectations: Bootstrap expects the navbar colors to be static (i.e. the same in light and dark mode). 

If we want to support different navbar colors we'll need to implement something that compliments the current setup.

This PR also sets `data-bs-theme="dark"` on the navbar `<nav>` element when `inverse = TRUE` and fixes the navbar toggle icon in dark mode.

> [!IMPORTANT]
> Re-test after merging #1134 

| Before | After | Description |
|:---:|:--:|:---|
| image | image | Dark mode with `inverse = FALSE`, black on light blue is expected but white toggle button icon is not. |
| image |  image | Dark mode with `inverse = TRUE`, white text on dark background is expected in both light/dark, before shows that light navbar bg was used in dark mode. |

<details>
<summary>Example app</summary>

```r
library(shiny)
pkgload::load_all()

ui <- page_navbar(
  title = "My Simple App",
  theme = bs_theme(
    preset = "default",
    navbar_bg = "#732400",
    navbar_light_bg = "#b2d8ff",
    navbar_dark_bg = "#00165d",
    cache_buster = format(as.integer(Sys.time()))
  ),
  inverse = TRUE,
  sidebar = sidebar(
    title = "Settings",
    "Just stuff here",
    input_dark_mode(),
  ),
  nav_panel("Home",
    h1("Welcome to the Home Page"),
    p("This is some boilerplate text for the home page."),
    card(
      card_header("Just a regular card"),
      layout_sidebar(
        sidebar = sidebar(
          "Some stuff here too",
          position = "right"
        ),
        "And card content with a sidebar!"
      )
    )
  )
)

server <- function(input, output, session) {

}

shinyApp(ui, server)
```

</details>

### CODE REVIEW

File: R/navs-legacy.R [hunk 1]
```diff
@@ -171,12 +178,12 @@ navs_bar_ <- function(..., title = NULL, id = NULL, selected = NULL,
...
-    # navbarPage_() returns a tagList() of the nav and content
-    navbar[[1]] <- tagAppendAttributes(
-      navbar[[1]], style = css(background_color = paste(bg, "!important"))
-    )
-  }
+  # navbarPage_() returns a tagList() of the nav and content
+  navbar[[1]] <- tagAppendAttributes(
+    navbar[[1]], 
+    style = if (!is.null(bg)) css(background_color = paste(bg, "!important")),
+    "data-bs-theme" = navbar_color_mode
```
Comment (cpsievert) [inline R/navs-legacy.R:185]: This way we could get rid of `navbar_color_mode`?

```suggestion
    "data-bs-theme" = if (inverse) "dark" else "light"
```

File: inst/bs3compat/_navbar_compat.scss [hunk 2]
```diff
@@ -101,15 +101,6 @@ ul.nav.navbar-nav {
...
 }
 
-$enable-dark-mode: false !default;
-@if $enable-dark-mode {
-  @include color-mode(dark) {
-    .navbar.navbar-default {
-      background-color: var(--bslib-navbar-default-bg, var(--#{$prefix}dark)) !important;
-    }
-  }
-}
```
Comment (cpsievert) [inline inst/bs3compat/_navbar_compat.scss:111]: Let's make sure to test with `version = 4` as well (and if it's needed for old version(s), we can bring in these rules conditionally based on `$bootstrap-version`

File: inst/lib/bs5/scss/_navbar.scss [hunk 3]
```diff
@@ -307,11 +307,17 @@
...
+  }
 }
 
 @if $enable-dark-mode {
   @include color-mode(dark) {
-    .navbar-toggler-icon {
+    // patched: toggler follows global theme unless in a light region
+    .navbar:not([data-bs-theme="light"]) .navbar-toggler-icon {
       --#{$prefix}navbar-toggler-icon-bg: #{escape-svg($navbar-dark-toggler-icon-bg)};
     }
```
Comment (cpsievert) [inline inst/lib/bs5/scss/_navbar.scss:322]: Is this gonna work sensibly for scenarios where we don't have control over the navbar markup (e.g., pkgdown, rmarkdown, flexdashboard, etc.)?

File: R/navs-legacy.R (hunk 1 above)
Comment (gadenbuie) [reply R/navs-legacy.R:185]: Part of the problem I'm trying to solve is that this isn't really a binary choice, at minimum we want to support three different use-cases:

1. A light background navbar with `data-bs-theme="light"`
2. A dark background navbar with `data-bs-theme="dark"`
3. A navbar that follows the global `data-bs-theme` setting, which we achieve by setting `data-bs-theme="auto"`.

We won't depend on `"auto"` directly; we'll rely on the fact that it's not a known color mode, in which case the global color mode can affect the navbar colors.

File: R/navs-legacy.R [hunk 4]
```diff
@@ -211,7 +218,7 @@ navbarPage_ <- function(title,
   position <- match.arg(position)
   if (!is.null(position))
     navbarClass <- paste0(navbarClass, " navbar-", position)
-  if (inverse)
+  if (isTRUE(inverse))
```
[the remark this answers is not in the document]
Comment (gadenbuie) [reply R/navs-legacy.R:221]: Yeah, e.g. `inverse = "auto"` (see above).

File: inst/bs3compat/_navbar_compat.scss (hunk 2 above)
Comment (gadenbuie) [reply inst/bs3compat/_navbar_compat.scss:111]: You added this in 

* https://github.com/rstudio/bslib/pull/764

... and BS4 doesn't have an `$enable-dark-mode` flag.

File: inst/lib/bs5/scss/_navbar.scss (hunk 3 above)
Comment (gadenbuie) [reply inst/lib/bs5/scss/_navbar.scss:322]: I can look into this and will add anything that needs to be addressed to https://github.com/posit-dev/brand-yml/issues/52. Bringing pkgdown and bslib back into alignment is part of that plan.

Comment (gadenbuie) [reply inst/lib/bs5/scss/_navbar.scss:322]: This whole PR does interact with rmarkdown websites and the approach I'd like to take is to make `.navbar-default` and `.navbar-inverse` be no-op classes for BS 5+. We could also take them out of the markup when we know that we're in BS 5+. That's going to require another pass at this PR, though.

**Comment by gadenbuie** [comment]
Closing this PR in favor of #1145, which builds on the work here and but includes recent PRs and an increased scope.

**Pull request closed without merging**